fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/LimitPruneTabletsRule.java

```diff
@@ -77,9 +77,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
             }
         }
 
-        // 1. Don't select any tablet
-        // 2. selected tablets don't change
-        if (result.isEmpty() || result.equals(olapScanOperator.getSelectedTabletId())) {
+        // 1. totalRow must larger than limit
+        // 2. Don't select any tablet
+        // 3. selected tablets don't change
+        if (totalRow < limit || result.isEmpty() || result.equals(olapScanOperator.getSelectedTabletId())) {
```

Comment:
If `totalRow < limit`, all the non-empty tablets are in `result`, why is it wrong?

Reply:
Is it because the meta may be old, so the real `tabletRowCount` is greater than `tablet.getRowCount(version)`?

Reply:
Add comment, report delay is real reason